### PR TITLE
Merge 06.07.15 develop

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

**Comment by livecodeali** [comment]
Once this has been merged, we should also merge https://github.com/runrev/livecode-ide/pull/344 and update the submodule pointer.

**Comment by peter-b** [comment]
@livecode-vulcan review ok f167531

Things to do when merged (summarizing from above):
- Merge runrev/livecode-ide#344 and update submodule pointer
- Add call to `sync_mfocus` to widget host (in `MCInterfaceExecCreateWidget()`)

:+1:

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok f167531ed1d1d951dd4a6fc74c1bd7a03eedc6f7

**Comment by livecode-vulcan** [comment]
😞 test failure f167531ed1d1d951dd4a6fc74c1bd7a03eedc6f7
- try-linux-x86: [failure](https://vulcan.livecode.com/builders/try-linux-x86/builds/419)
- try-win-x86: [pending](https://vulcan.livecode.com/builders/try-win-x86/builds/479)
- try-ios: [pending](https://vulcan.livecode.com/builders/try-ios/builds/365)
- try-android-armv6: [failure](https://vulcan.livecode.com/builders/try-android-armv6/builds/430)
- try-linux-x86_64: [failure](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/365)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/356)

**Comment by livecodeali** [comment]
OK, I have amended that last commit to remove a comment which is no longer relevant. This is as a result of a change which also makes
- Merge runrev/livecode-ide#344 and update submodule pointer

no longer necessary.

The following 
- Add call to `sync_mfocus` to widget host (in `MCInterfaceExecCreateWidget()`) 

has been dealt with in commit 7a17f71 above.

So we just need to work out why it doesn't build on linux :confused:

**Comment by peter-b** [comment]
@livecodeali According to @livecodefraser the build failures are because the CEF version in the fetch-libraries.sh script has got clobbered.

File: prebuilt/fetch-libraries.sh [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #!/bin/bash
 
 # Library versions
-VERSION_OpenSSL="1.0.1m"
-VERSION_Curl="7.21.1"
-VERSION_ICU="52.1"
-VERSION_CEF="39.0.2171.95"
```
Comment (peter-b) [inline prebuilt/fetch-libraries.sh:7]: This line shouldn't have been changed (`VERSION_CEF` should still be `39.0.2171.95`)

File: prebuilt/package-libs.sh [hunk 2]
```diff
@@ -1,10 +1,10 @@
 #!/bin/bash
 
 # Versions
-OPENSSL_VERSION="1.0.1m"
-CURL_VERSION="7.21.1"
-ICU_VERSION="52.1"
-CEF_VERSION="39.0.2171.95"
```
Comment (peter-b) [inline prebuilt/package-libs.sh:7]: Same `CEF_VERSION` issue.

**Comment by peter-b** [comment]
@livecode-vulcan review ok 27fdfb0

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 27fdfb0e0f881dffccf6d0e28cc16e29d215662f

**Comment by livecode-vulcan** [comment]
😞 test failure 27fdfb0e0f881dffccf6d0e28cc16e29d215662f
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/428)
- try-win-x86: [pending](https://vulcan.livecode.com/builders/try-win-x86/builds/488)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/374)
- try-android-armv6: [failure](https://vulcan.livecode.com/builders/try-android-armv6/builds/439)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/374)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/365)

**Comment by peter-b** [comment]
Okay, so the problem _now_ is that `-llog` only needs to be passed for **target** builds.

**Comment by livecodeali** [comment]
so...

```
			'link_settings':
			{
				'conditions':
				[
					[
						'OS == "mac" or OS == "ios"',
						{
							'libraries':
							[
								'$(SDKROOT)/System/Library/Frameworks/CoreFoundation.framework',
								'$(SDKROOT)/System/Library/Frameworks/Foundation.framework',
							],
						},
					],
					[
						'OS == "android"',
						{
							'libraries':
							[
								'-lm',
							],
						},
					],
					[
						'_toolset == "target"',
						{
							'libraries':
							[
								'-llog',
							],
						},
					],
				],
			},
```
?

**Comment by livecodeali** [comment]
So...

```
            'link_settings':
            {
                'conditions':
                [
                    [
                        'OS == "mac" or OS == "ios"',
                        {
                            'libraries':
                            [
                                '$(SDKROOT)/System/Library/Frameworks/CoreFoundation.framework',
                                '$(SDKROOT)/System/Library/Frameworks/Foundation.framework',
                            ],
                        },
                    ],
                    [
                        'OS == "android"',
                        {
                            'libraries':
                            [
                                '-lm',
                            ],
                        },
                    ],
                    [
                        'OS' == "android" and '_toolset == "target"',
                        {
                            'libraries':
                            [
                                '-llog',
                            ],
                        },
                    ],
                ],
            },
```

?

**Comment by peter-b** [comment]
That looks plausible. Paging @livecodefraser as our resident gyp expert. :wink:

**Comment by livecodefraser** [comment]
That does indeed look plausible...

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay b47c3ef

Time to find out if my built-in Gyp parser matches the real one...

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok b47c3efda0835a2f6b78e63035d4127ee2fc8cb8

**Comment by livecodeali** [comment]
I did tweak it slightly, as 
`'OS' == "android" and '_toolset == "target"',` 
doesn't have the quotes quite right. Should be
`'OS == "android" and _toolset == "target"',`

**Comment by livecode-vulcan** [comment]
😞 test failure b47c3efda0835a2f6b78e63035d4127ee2fc8cb8
- try-linux-x86: [failure](https://vulcan.livecode.com/builders/try-linux-x86/builds/434)
- try-win-x86: [failure](https://vulcan.livecode.com/builders/try-win-x86/builds/494)
- try-ios: [failure](https://vulcan.livecode.com/builders/try-ios/builds/380)
- try-android-armv6: [failure](https://vulcan.livecode.com/builders/try-android-armv6/builds/445)
- try-linux-x86_64: [failure](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/380)
- try-mac: [failure](https://vulcan.livecode.com/builders/try-mac/builds/371)

**Comment by peter-b** [comment]
@livecode-vulcan reviewed by @livecodefraser ok 7325dfe

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 7325dfe

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 7325dfe40d444340030a3699ab38da3f4014c1e3

**Comment by livecode-vulcan** [comment]
😞 test failure 7325dfe40d444340030a3699ab38da3f4014c1e3
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/436)
- try-win-x86: [pending](https://vulcan.livecode.com/builders/try-win-x86/builds/496)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/382)
- try-android-armv6: [failure](https://vulcan.livecode.com/builders/try-android-armv6/builds/447)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/382)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/373)

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 337cdce

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 337cdce96007480bb18b6e159ac986b8e4744e36

**Comment by livecode-vulcan** [comment]
😎 test success 337cdce96007480bb18b6e159ac986b8e4744e36
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/442)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/502)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/388)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/453)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/388)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/379)

**Pull request merged**